goaviatrix/controller_bgp_max_as_limit_config.go

```diff
@@ -0,0 +1,74 @@
+package goaviatrix
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+func (c *Client) SetControllerBgpMaxAsLimit(ctx context.Context, maxAsLimit int) error {
+	data := map[string]string{
+		"action":       "set_bgp_max_as_limit",
+		"CID":          c.CID,
+		"max_as_limit": fmt.Sprint(maxAsLimit),
+	}
+
+	checkFunc := func(action, reason string, ret bool) error {
+		if !ret && !strings.Contains(reason, "Configured BGP maximum AS limit is not changed") {
+			return fmt.Errorf("rest API %s Post failed: %s", action, reason)
+		}
+		return nil
+	}
+	return c.PostAPIContext(ctx, data["action"], data, checkFunc)
+}
+
+func (c *Client) DisableControllerBgpMaxAsLimit(ctx context.Context) error {
+	data := map[string]string{
+		"action":       "set_bgp_max_as_limit",
+		"CID":          c.CID,
+		"max_as_limit": "",
+	}
+
+	checkFunc := func(action, reason string, ret bool) error {
+		if !ret && !strings.Contains(reason, "Configured BGP maximum AS limit is not changed") {
+			return fmt.Errorf("rest API %s Post failed: %s", action, reason)
+		}
+		return nil
+	}
+	return c.PostAPIContext(ctx, data["action"], data, checkFunc)
+}
+
+func (c *Client) GetControllerBgpMaxAsLimit(ctx context.Context) (int, error) {
+	data := map[string]string{
+		"action": "show_bgp_max_as_limit",
+		"CID":    c.CID,
+	}
+
+	type BgpMaxAsLimitResponse struct {
+		Return bool   `json:"return"`
+		Reason string `json:"reason"`
+		Result string `json:"results"`
+	}
+
+	var resp BgpMaxAsLimitResponse
+	err := c.GetAPIContext(ctx, &resp, data["action"], data, BasicCheck)
+	if err != nil {
+		return -1, err
```

Comment:
In Go, it is standard to just return the zero value for a type if there is an error. For `int` that would just be `0`